etc/completions/flux.pre

```diff
@@ -9,56 +9,963 @@
+    link_OPTS="\
+        -N --namespace= \
+        -O --treeobj-root \
+        -b --blobref \
+        -s --sequence \
+    "
+    readlink_OPTS="\
+        -N --namespace= \
+        -a --at= \
+        -o --namespace-only \
+        -k --key-only \
+    "
+    mkdir_OPTS="\
+        -N --namespace= \
+        -O --treeobj-root \
+        -b --blobref \
+        -s --sequence \
+    "
+    copy_OPTS="\
+        -S --src-namespace= \
+        -D --dst-namespace= \
+    "
+    move_OPTS="\
+        ${copy_OPTS}
+    "
+    dropcache_OPTS="\
+        -a --all
+    "
+    version_OPTS="\
+        -N --namespace= \
+    "
+    wait_OPTS="\
+        -N --namespace= \
+    "
+    getroot_OPTS="\
+        -N --namespace= \
+        -s --sequence \
+        -o --owner \
+        -b --blobref \
+    "
+    if [[ $cmd != "kvs" ]]; then
+
+        if _flux_contains_word "eventlog" ${COMP_WORDS[*]}; then
+            _flux_kvs_eventlog $cmd
+            return 0
         fi
-    done
 
-    for builtin in $FLUX_BUILTINS; do
-        compopts+="$builtin "
-    done
+        var="${cmd//-/_}_OPTS"
+        COMPREPLY=( $(compgen -W "${!var}" -- "$cur") )
+        if [[ "${COMPREPLY[@]}" == *= ]]; then
+            # Add space if there is not a '=' in suggestions
+            compopt -o nospace
+        fi
+    else
+        COMPREPLY=( $(compgen -W "${subcmds}" -- "$cur") )
+    fi
+}
+
+# flux-overlay(1) completions
+_flux_overlay()
+{
+    local cmd=$1
+    local subcmds="status lookup parentof disconnect"
+
+    local status_OPTS="\
+        -h --help \
+        -r --rank= \
+        -v --verbose= \
+        -t --timeout= \
+        --summary \
+        --down \
+        --no-pretty \
+        --no-ghost \
+        -L --color= \
+        -H --highlight= \
+        -w --wait= \
+    "
+    local lookup_OPTS="\
+        -h --help \
+    "
+    local parentof_OPTS="\
+        -h --help \
+    "
+    local disconnect_OPTS="\
+        -h --help \
+        -r --parent= \
+    "
+    if [[ $cmd != "overlay" ]]; then
+        var="${cmd//-/_}_OPTS"
+        COMPREPLY=( $(compgen -W "${!var}" -- "$cur") )
+        if [[ "${COMPREPLY[@]}" == *= ]]; then
+            # Add space if there is not a '=' in suggestions
+            compopt -o nospace
+        fi
+    else
+        COMPREPLY=( $(compgen -W "${subcmds}" -- "$cur") )
+    fi
+    return 0
+}
+
+# flux-config(1) completions
+_flux_config()
+{
+    local cmd=$1
+    local subcmds="reload get builtin"
+
+    local get_OPTS="\
+        -t --type= \
+        -q --quiet \
+        -d --default= \
+    "
+
+    if [[ $cmd != "config" ]]; then
+        var="${cmd//-/_}_OPTS"
+        COMPREPLY=( $(compgen -W "${!var} -h --help" -- "$cur") )
+        if [[ "${COMPREPLY[@]}" == *= ]]; then
+            # Add space if there is not a '=' in suggestions
+            compopt -o nospace
+        fi
+    else
+        COMPREPLY=( $(compgen -W "${subcmds}" -- "$cur") )
+    fi
+    return 0
+}
+
+# flux-admin completions
```

Comment:
flux-admin(8) for consistency of other comments

Reply:
oh wait, seems to be installed in /usr/bin?  is that what we want?

Reply:
> oh wait, seems to be installed in /usr/bin? is that what we want?

I'm not 100% sure what this is referring to, but it doesn't sound right :slightly_smiling_face: Want to explain more?

Reply:
Yeah, there's no man page for `flux-admin` but I put `flux-admin(8)` anyway. Note that `flux-admin.py` should not be in `/usr/bin` but `/usr/libexec/flux/cmd/flux-admin.py`.